protocol-units/mempool/util/src/lib.rs

```diff
@@ -1,172 +1,224 @@
 use serde::{Deserialize, Serialize};
 
-use movement_types::{Block, Transaction, Id};
+use movement_types::{Block, Id, Transaction};
 use std::cmp::Ordering;
+use thiserror::Error;
+
+pub type BoxedStoreError = Box<dyn std::error::Error + Send + Sync + 'static>;
+
+#[derive(Error, Debug)]
+pub enum MempoolTransactionOperationsError {
+	#[error("Serialization error: {0}")]
+	SerializationError(String),
+	#[error("Deserialization error: {0}")]
+	DeserializationError(String),
+	#[error("Underlying store error")]
+	StoreError(#[from] BoxedStoreError),
+	#[error("Mock error: {0}")]
+	MockError(String),
```

Comment:
Why expose mock errors specially as part of the API?
The mock is supposed to be a stand-in for the real component and as such should not generate any special errors or code paths, should it?

Reply:
Correct, placed it in to `#[cfg(test)]` it out in a later stadium.

Reply:
We do not want to have error variants in the public API. One solution is to use the "Other" variant as an error variant, which can be utilized in cases where one wishes to return a string error. Pushed it and consider it as resolved.